### ISSUE TITLE
Feature Suggestion: Download Agent (stage1) in small parts to hide log size detections

### ISSUE THREAD

**Issue by ceramicskate0** [EmpireProject/Empire#1115]
## Empire Version
future version

## OS Information (Linux flavor, Python version)
Linux

## Expected behavior and description of the error, including any actions taken immediately prior to the error. The more detail the better.
No error.

## Screenshot of error, embedded text output, or Pastebin link to the error
none

## Any additional information
At an event a detection for empire was shown to me based on counting char's length of Powershell cmdline.  The detection was effective at killing agents during the event. The idea occurred to me to either have the option or do by default to have stage 1 (i think its stage 1) download the agent code in small parts storing it on the machine in a temp location (maybe env variable or reg key) until it gets the whole payload. then executes the contents of that location and clears location. Then moves onto stage 2.  Im thinking It will kick off a lot of powershell executions but should have smaller cmdline logs.

**Comment by passwordleak** [comment]
Are you referring to the fact they're blocking long powershell commands? I have noticed some AVs will block long powershell commands. 

You can use something like https://github.com/danielbohannon/Invoke-CradleCrafter

Also, have you tried using the launcher_xml (msbuild) payload?

**Comment by ceramicskate0** [comment]
Yes. the cradles and other LOLBINs work great. I noticed a 'trend' of simply count of the characters of event log command line >x chars(ex 200) and/or a lack of space char's finds all the empire stuff . A grwoing number of organizations are starting to think this is a 90% solution to detection. I was wondering if this could be challenged though automation of the framework? I agree with invoke obfuscation already integrated invoke cradle crafter shouldn't take as long (possible famous last words) and would be great to see in the framework.

**Comment by ceramicskate0** [comment]
I was thinking of possibly putting the contents in environment variable until execution then clearing contents. Thoughts?
